MekHQ/src/mekhq/gui/adapter/UnitTableMouseAdapter.java

```diff
@@ -146,13 +146,32 @@
     //endregion Commands
     //endregion Variable Declarations
 
-    public UnitTableMouseAdapter(CampaignGUI gui, JTable unitTable, UnitTableModel unitModel) {
-        super();
+    protected UnitTableMouseAdapter(CampaignGUI gui, JTable unitTable, UnitTableModel unitModel) {
         this.gui = gui;
         this.unitTable = unitTable;
         this.unitModel = unitModel;
     }
 
+    public static void connect(CampaignGUI gui, JTable unitTable, UnitTableModel unitModel, JSplitPane splitUnit) {
+        new UnitTableMouseAdapter(gui, unitTable, unitModel) {
+            @Override
+            public void mouseClicked(MouseEvent e) {
+                if (e.getButton() == MouseEvent.BUTTON1 && e.getClickCount() == 2) {
+                    int width = splitUnit.getSize().width;
+                    int location = splitUnit.getDividerLocation();
+                    int size = splitUnit.getDividerSize();
+                    if ((width - location + size) < HangarTab.UNIT_VIEW_WIDTH) {
+                        // expand
+                        splitUnit.resetToPreferredSizes();
+                    } else {
+                        // collapse
+                        splitUnit.setDividerLocation(1.0);
+                    }
+                }
+            }
+        }.connect(unitTable);
```

Comment:
Same as personnel, why are we not just overriding this is the class given it is only used there?

Reply:
This is so I do not have to include the splitter as a variable in the class.